packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ApiKeyAuthClientProviderTests.cs

```diff
@@ -0,0 +1,457 @@
+            var primaryCtorBody = primaryPublicConstructor?.BodyStatements;
+            Assert.IsNotNull(primaryCtorBody);
+        }
+
+        private void ValidateSecondaryConstructor(
+            ConstructorProvider? primaryConstructor,
+            ConstructorProvider? secondaryPublicConstructor,
+            List<InputParameter> inputParameters)
+        {
+            Assert.IsNotNull(secondaryPublicConstructor);
+            var ctorParams = secondaryPublicConstructor?.Signature?.Parameters;
+
+            // secondary ctor should consist of all required parameters + auth parameter
+            var requiredParams = inputParameters.Where(p => p.IsRequired).ToList();
+            Assert.AreEqual(requiredParams.Count + 1, ctorParams?.Count);
+            var endpointParam = ctorParams?.FirstOrDefault(p => p.Name == KnownParameters.Endpoint.Name);
+
+            if (requiredParams.Count == 0)
+            {
+                // auth should be the only parameter if endpoint is optional
+                Assert.AreEqual("keyCredential", ctorParams?[0].Name);
+            }
+            else
+            {
+                // otherwise, it should only consist of the auth parameter
+                Assert.AreEqual(KnownParameters.Endpoint.Name, ctorParams?[0].Name);
+                Assert.AreEqual("keyCredential", ctorParams?[1].Name);
+            }
+
+            Assert.AreEqual(MethodBodyStatement.Empty, secondaryPublicConstructor?.BodyStatements);
+
+            // validate the initializer
+            var initializer = secondaryPublicConstructor?.Signature?.Initializer;
+            Assert.AreEqual(primaryConstructor?.Signature?.Parameters?.Count, initializer?.Arguments?.Count);
+        }
+
+        [TestCaseSource(nameof(EndpointParamInitializationValueTestCases))]
+        public void EndpointInitializationValue(InputParameter endpointParameter, ValueExpression? expectedValue)
+        {
+            var client = InputFactory.Client(TestClientName, parameters: [endpointParameter]);
+            var clientProvider = new ClientProvider(client);
+
+            Assert.IsNotNull(clientProvider);
+            // find the endpoint parameter from the primary constructor
+            var primaryConstructor = clientProvider.Constructors.FirstOrDefault(
+                c => c.Signature?.Initializer == null && c.Signature?.Modifiers == MethodSignatureModifiers.Public);
+            var endpoint = primaryConstructor?.Signature?.Parameters?.FirstOrDefault(p => p.Name == KnownParameters.Endpoint.Name);
+
+            Assert.IsNotNull(endpoint);
+            Assert.AreEqual(expectedValue?.GetType(), endpoint?.InitializationValue?.GetType());
+            if (expectedValue != null)
+            {
+                Assert.IsTrue(endpoint?.InitializationValue is NewInstanceExpression);
+            }
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void TestGetClientOptions(bool isSubClient)
```

Comment:
does this need to be scoped to a specific type of auth client? It seems this will be common amongst all clients and can instead go in `ClientProviderTests` if I'm understanding your proposed structure correctly. There are other test cases in this file that I think fit this pattern.

Reply:
Yeah this is a mistake that I took it here. I will put it back